src/main/java/org/folio/service/pieces/flows/update/PieceUpdateFlowManager.java

```diff
@@ -92,10 +108,49 @@ public Future<Void> updatePiece(Piece pieceToUpdate, boolean createItem, boolean
   }
 
   protected Future<Void> updatePoLine(PieceUpdateHolder holder, RequestContext requestContext) {
-    if (!Boolean.TRUE.equals(holder.getOriginPoLine().getIsPackage()) && !Boolean.TRUE.equals(holder.getOriginPoLine().getCheckinItems())) {
-      return updatePoLineService.updatePoLine(holder, requestContext);
-    }
-    return Future.succeededFuture();
+    CompositePoLine originPoLine = holder.getOriginPoLine();
+
+    return getPiecesByPoLine(originPoLine.getId(), requestContext)
+      .compose(pieces -> {
+        CompositePurchaseOrder order = holder.getOriginPurchaseOrder();
+        if (order.getOrderType() != OrderType.ONE_TIME || order.getWorkflowStatus() != WorkflowStatus.OPEN) {
+          return Future.succeededFuture();
+        }
+        logger.info(">>> " + pieces);
+        List<Piece> piecesToUpdate = List.of(holder.getPieceToUpdate());
+        logger.info(">>> " + calculatePoLineReceiptStatus(originPoLine, pieces, piecesToUpdate));
+        return Future.succeededFuture();
+      }).compose(t -> {
```

Comment:
this method should be refactored, this if statement can return Future.succeededFuture()
 if (order.getOrderType() != OrderType.ONE_TIME || order.getWorkflowStatus() != WorkflowStatus.OPEN) 

and in this case we should not do these checks for 
originPoLine.getIsPackage() and originPoLine.getCheckinItems(), we just need to return from method

Reply:
fixed